__tests__/app.test.js

```
@@ -295,6 +295,86 @@ describe("/api/articles", () => {
+        .get("/api/articles?sort_by=author")
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy("author", {
+            descending: false,
+          });
+        });
+    });
+    it("200: allows the user to sort articles by any valid column (votes, default descending)", () => {
+      return request(app)
+        .get("/api/articles?sort_by=votes")
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy("votes", {
+            descending: true,
+          });
+        });
+    });
+    it("200: allows the user to sort articles by any valid column (created_at, default descending)", () => {
+      return request(app)
+        .get("/api/articles?sort_by=created_at")
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy("created_at", {
+            descending: true,
+          });
+        });
+    });
+    it("200: allows user to choose whether the order is ascending or descending", () => {
+      return request(app)
+        .get("/api/articles?sort_by=votes&&order=asc")
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy("votes", {
+            descending: false,
+          });
+        });
+    });
+    it("405: only allows greenlisted sortby and order queries", () => {
+      return request(app)
+        .get("/api/articles?sort_by=bananas")
+        .expect(405)
+        .then(({ body: { message } }) => {
+          expect(message).toBe("405 - Method Not Allowed");
+        });
+    });
+    it("200: defaults to descending (alphabetical ascending) when invalid order given", () => {
+      return request(app)
+      .get("/api/articles?sort_by=author&&order=anything_I_want")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("author", {
+          descending: false
+        });
+      });
```

Please throw an error is the order by is invalid.